src/carousel.js

```diff
@@ -94,11 +103,25 @@ class Carousel extends Component {
     scrollDistance < 0
```

Comment:
Let's say that `scrollDistance < 0`, `this.currentIndex===1` and `this.alwaysCentralizeSelected===true`. `this.scrollToIndex(0)` will be called and will change `this.currentIndex` to `0`. Then `this.scrollToIndex(1)` will be called.
`scrollToIndex` should be called only once when drag ends...

Reply:
this was an intended behavior in order to display an automatic (drag back effect).
but i agree, it's not the best possible impl.

now it's fixed by adding the following validation to `scrollToIndex` 
`if(this.alwaysCentralizeSelected  && (index === 0 || index === this.data.length-1)) 
return;`